include/alpaka/test/event/EventHostManualTrigger.hpp

```diff
@@ -437,7 +434,17 @@ namespace alpaka::trait
             //   on host updates may hang. This includes synchronization between the host and
             //   the device build upon value-based CUDA queue synchronization APIs such as
             //   cuStreamWaitValue32() and cuStreamWriteValue32().
-            ALPAKA_CUDA_DRV_CHECK(cuStreamWaitValue32(
+            using StreamWaitValueFunc = CUresult (*)(CUstream, CUdeviceptr, cuuint32_t, unsigned int);
+#    if(CUDA_VERSION < 11070) || (CUDA_VERSION >= 12000)
+            // NVIDIA introduced a new stream memory ops API with CUDA 11.7 (called v2). The corresponding CUDA
+            // functions were suffixed with `_v2`. With CUDA 12.0 v1 of the API was removed and the `_v2` removed
+            // from the new functions. So CUDA <= 11.6 and CUDA >= 12.0 share the same function signature but
+            // internally do different things.
+            auto streamWaitValue = StreamWaitValueFunc{cuStreamWaitValue32};
+#    else
+            auto streamWaitValue = StreamWaitValueFunc{cuStreamWaitValue32_v2};
+#    endif
```

Comment:
Could you please refactor this into a separate function, so the #if logic and comment is not duplicated 3 times? Thx!

Reply:
Done,